### PR TITLE
azure backend

### PR DESCRIPTION
**Pull request by nitely**
**DO NOT MERGE!**

This is a complete overhaul of the current azure backend. The plan is to get the master branch up-to-date with upstream before merging this.

I think there are some projects (on prod) depending on this branch, so we've to update them to depend on some other branch based on master. Not sure how you want to handle it @guydou @gilgamezh 

All of this so https://github.com/jschneier/django-storages/pull/318 gets updated, but maybe I should just open a new PR...

### CODE REVIEW

**Comment by nitely** [comment]
meh, I think the better way is to not touch master for the time being. Projects that need these changes can just depend on `nitely_azure` branch. Thoughts?

**Comment by gilgamezh** [comment]
+1 we can use this branch while we wait to merge in upstream

File: storages/backends/azure_storage.py [hunk 1]
```diff
@@ -1,132 +1,170 @@
...
-    buffer_size = setting('AZURE_FILE_BUFFER_SIZE', 4194304)
-
-    def __init__(self, *args, **kwargs):
-        super(AzureStorage, self).__init__(*args, **kwargs)
+    upload_max_conn = setting("AZURE_UPLOAD_MAX_CONN", 2)
+    timeout = setting('AZURE_CONNECTION_TIMEOUT_SECS', 20)
+    max_memory_size = setting('AZURE_BLOB_MAX_MEMORY_SIZE', 2*1024*1024)
+    expiration_secs = setting('AZURE_URL_EXPIRATION_SECS')
+    overwrite_files = setting('AZURE_OVERWRITE_FILES', True)
+    _location = setting('AZURE_LOCATION', '')
```
Comment (akshaybabloo) [inline storages/backends/azure_storage.py:145]: Why `_location` and not `location`?

Comment (nitely) [reply storages/backends/azure_storage.py:145]: I guess I thought it'd be more clear/clean having the class var as "private" and the attribute public. But now that I know users will actually subclass this, it'd be better to leave it public. Good catch! thanks! I'll update the `nitely_azure` branch. Edit: actually, I'm not sure why there's a location attribute, I think there was another backend doing it this way. I'll look into it later.

**Comment by nitely** [comment]
ok, this brach is too outdated now, use `nitely_azure` please.

**Comment by akshaybabloo** [comment]
Any update on adding a pull request of `nitely_master` to `jschneier/django-storages` (https://github.com/jschneier/django-storages/issues/532)

**Pull request closed without merging**